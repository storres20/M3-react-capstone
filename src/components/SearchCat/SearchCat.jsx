/* eslint-disable max-len */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styles from './SearchCat.module.css';
// Redux
import { updateCategory } from '../../redux/movies/categories';
// ***

function SearchCat() {
  /* Load Redux State */
  const allCategory = useSelector((state) => state.allCategory);

  /* Setup Redux dispatch */
  const dispatch = useDispatch();

  /* allCategory = movies (original values) */

  // Filter search bar
  const handleSearch = (event) => {
    const keyword = event.target.value;

    if (keyword !== '') {
      const results = allCategory.filter((user) => user.name.toLowerCase().includes(keyword.toLowerCase()));

      dispatch(updateCategory(results));
    } else {
      dispatch(updateCategory(allCategory));
    // If the text field is empty, show all users
    }
  };
  return (
    <div>
      <div className={styles.divFlex}>
        <input className={styles.inputFlex} type="text" placeholder="Search by category..." onChange={(event) => handleSearch(event)} />
      </div>
    </div>
  );
}

export default SearchCat;
