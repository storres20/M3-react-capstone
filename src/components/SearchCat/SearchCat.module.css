.divFlex {
  display: flex;
  justify-content: center;
}

.inputFlex {
  background-color: black;
  color: white;
  max-width: 300px;
  min-width: 250px;
  margin-top: 30px;
  height: 40px;
  border-radius: 8px;
}
