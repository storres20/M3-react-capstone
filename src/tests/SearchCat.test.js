import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import categoriesReducer from '../redux/movies/categories';
import allCategoryReducer from '../redux/movies/allCategory';
import SearchCat from '../components/SearchCat/SearchCat';

// configureStore
const createTestStore = () => {
  const store = configureStore({
    reducer: {
      category: categoriesReducer,
      allCategory: allCategoryReducer,
    },
  });

  return store;
};
export default createTestStore;

// Test
describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    // Create a redux store
    component = renderer.create(
      <Provider store={store}>
        <SearchCat />
      </Provider>,
    ).toJSON();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
