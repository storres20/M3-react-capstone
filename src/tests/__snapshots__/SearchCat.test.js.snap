// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div
  className="divFlex"
  data-testid="SearchCat"
>
  <input
    className="inputFlex"
    onChange={[Function]}
    placeholder="Search by category..."
    type="text"
  />
</div>
`;
